arch/arm/core/aarch32/swap_helper.S

```diff
@@ -71,6 +71,10 @@ SECTION_FUNC(TEXT, z_arm_pendsv)
     ldr r1, =_kernel
     ldr r2, [r1, #_kernel_offset_to_current]
 
+#if defined(CONFIG_ARM_STORE_EXC_RETURN)
```

Comment:
Please, add a comment:
E.g. `Store LSB of LR (EXC_RETURN) to the thread's mode octet reserved for this.`

Reply:
looks fine